### PR TITLE
stanford-parser-4.2.0

### PR DESCRIPTION
**Pull request by mattmoore**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/stanford-parser.rb [hunk 1]
```diff
@@ -1,10 +1,8 @@
 class StanfordParser < Formula
   desc "Statistical NLP parser"
   homepage "https://nlp.stanford.edu/software/lex-parser.shtml"
-  url "https://nlp.stanford.edu/software/stanford-parser-full-2018-10-17.zip"
-  version "3.9.2"
-  sha256 "92d852af54c0727c2367b9ce267c53bf08f1551a08ec5dd92c357b8cc7b2bcd9"
-  revision 1
+  url "https://nlp.stanford.edu/software/stanford-parser-4.2.0.zip"
+  sha256 "8c2110c78f7f82b66bcf91089a18e415669eda4346bbd9a6e3bc2bde63e5fed1"
```
Comment (carlocab) [inline Formula/stanford-parser.rb:5]:
```suggestion
  sha256 "8c2110c78f7f82b66bcf91089a18e415669eda4346bbd9a6e3bc2bde63e5fed1"
  license "GPL-2.0-or-later"
```
From `README.txt`:
```
// This program is free software; you can redistribute it and/or
// modify it under the terms of the GNU General Public License
// as published by the Free Software Foundation; either version 2
// of the License, or (at your option) any later version.
```
If you make this change, squash your commits together, please. Thanks!

Comment (mattmoore) [reply Formula/stanford-parser.rb:5]: Sounds good!

**Comment by mattmoore** [comment]
I noticed the test fails because of JVM memory settings in the upstream source from Stanford, which is fixed when bumping the test from `150m` to `200m`. I've created a PR https://github.com/stanfordnlp/CoreNLP/pull/1140 to address this, but don't know when it would be merged and available in their latest binary distribution. So might make sense to either put this PR on hold, or remove it until that's addressed?

**Comment by mattmoore** [comment]
> I noticed the test fails because of JVM memory settings in the upstream source from Stanford, which is fixed when bumping the setting in their launcher from `150m` to `200m`. I've created a PR [stanfordnlp/CoreNLP#1140](https://github.com/stanfordnlp/CoreNLP/pull/1140) to address this, but don't know when it would be merged and available in their latest binary distribution. So might make sense to either put this PR on hold, or remove it until that's addressed?

Nevermind. I ended up finding another sample they had included `english-onesent.txt` that is small enough to fit. So I've made that change and test is passing locally now.

**Review by carlocab — APPROVED** [review]

**Pull request merged**